app/Jobs/Brevo/ProcessBrevoWebhook.php

```diff
@@ -0,0 +1,471 @@
+
+        $data = array_merge($this->request, ['history' => $this->fetchMessage()]);
+
+        $sl = $this->getSystemLog($this->request['message-id']);
+
+        if ($sl) {
+            $this->updateSystemLog($sl, $data);
+            return;
+        }
+
+        (new SystemLogger($data, SystemLog::CATEGORY_MAIL, SystemLog::EVENT_MAIL_SPAM_COMPLAINT, SystemLog::TYPE_WEBHOOK_RESPONSE, $this->invitation->contact->client, $this->invitation->company))->handle();
+
+        if (config('ninja.notification.slack')) {
+            $this->invitation->company->notification(new EmailSpamNotification($this->invitation->company->account))->ninja();
+        }
+    }
+
+    private function discoverInvitation(string $message_id)
+    {
+        $invitation = false;
+
+        if ($invitation = InvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'invoice';
+            return $invitation;
+        } elseif ($invitation = QuoteInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'quote';
+            return $invitation;
+        } elseif ($invitation = RecurringInvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'recurring_invoice';
+            return $invitation;
+        } elseif ($invitation = CreditInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'credit';
+            return $invitation;
+        } elseif ($invitation = PurchaseOrderInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'purchase_order';
+            return $invitation;
+        } else {
+            return $invitation;
+        }
+    }
+
+    public function getRawMessage(string $message_id)
+    {
+
+        $Brevo = new TransactionalEmailsApi(null, Configuration::getDefaultConfiguration()->setApiKey('api-key', config('services.brevo.key')));
+        $messageDetail = $Brevo->getTransacEmailContent($message_id);
+        return $messageDetail;
+
+    }
+
+
+    public function getBounceId(string $message_id): ?int
+    {
+
+        $messageDetail = $this->getRawMessage($message_id);
+
+
+        $event = collect($messageDetail->getEvents())->first(function ($event) {
+
+            return $event?->Details?->BounceID ?? false;
+
+        });
+
+        return $event?->Details?->BounceID ?? null;
+
+    }
+
+    // TODO
+    private function fetchMessage(): array
+    {
+        if (strlen($this->request['message-id']) < 1) {
+            return $this->default_response;
+        }
+
+        try {
+
+            $messageDetail = $this->getRawMessage($this->request['message-id']);
+
+            $recipients = $this->request["email"];
+            $subject = $messageDetail->getSubject() ?? '';
+
+            $events = collect($messageDetail->getEvents())->map(function (GetTransacEmailContentEvents $event) { // @turbo124 event does only contain name & time property, how to handle transformation?!
```

Comment:
@turbo124 event does only contain name & time property, how to handle transformation?!

Reply:
@turbo124 any ideas?

Reply:
Judging by the webhook response, you do have access to the subject, and by a quick glance, it looks like most fields can be resolved.

Reply:
@turbo124 this question was just related to the event-properties.

Reply:
@paulwer 

I can see that brevo have many different types of event classes, you'll want to find the one that matches as close as possible the postmark one:

```
                        'bounce_id' => $event?->Details?->BounceID ?? '',
                        'recipient' => $event->Recipient ?? '',
                        'status' => $event->Type ?? '',
                        'delivery_message' => $event->Details->DeliveryMessage ?? $event->Details->Summary ?? '',
                        'server' => $event->Details->DestinationServer ??  '',
                        'server_ip' => $event->Details->DestinationIP ?? '',
                        'date' => \Carbon\Carbon::parse($event->ReceivedAt)->format('Y-m-d H:i:s') ?? '',
```

Reply:
@turbo124 the according class of brevo only provides an event-name and a timestamp. no bounce_id, recipient, status etc.

should this fields then be empty?

Reply:
You may want to look at this:

https://github.com/getbrevo/brevo-php/blob/main/docs/Model/GetInboundEmailEventsByUuid.md

This should provide most of the data that is required.

Reply:
@turbo124 Thanks, I also saw this, but I think this is for inbound emails and not transactional sended mails.

in case of transactional senden emails it would be this:
https://github.com/getbrevo/brevo-php/blob/main/docs/Model/GetTransacEmailContentEvents.md

Am I missing something?

Reply:
As of the design of brevo, there is a transaction for each email, so we can concatinate these data partly into the events data and ignore other properties?

Reply:
When we get the initial webhook, we do have a large quantity of data thou, yes? 

https://github.com/getbrevo/brevo-php/blob/main/docs/Model/GetEmailEventReportEvents.md

Reply:
yes, but none of which is historical.
or do you mean, we should reuse them?

Reply:
@paulwer 

That is correct, only the initial message is required for the subject / recipient.

In the events section the most important key is the event type (delivered / opened / spam etc etc)

If the recipient can be derived, that would also be handy in case multiple users are sent a message for a single invoice.

Reply:
@turbo124 please verify, this aproach :)